### ISSUE TITLE
[ISSUE] Issue with `databricks_metastore_assignment` resource

### ISSUE THREAD

**Issue by sivadotblog** [databricks/terraform-provider-databricks#2657]
Creating an issue based of Issue no 2095 https://github.com/databricks/terraform-provider-databricks/issues/2095- This is still not resolved.

Based on the new version ~>1.24.0, I expect the Metastore assignment to work using the Account level API. It does not work as expected.

### Configuration
```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.67.0"
    }
    databricks = {
      source  = "databricks/databricks"
      version = "1.24.1"
    }
  }
}

provider "databricks" {
  alias               = "uc-account"
  host                = var.dbk-account-host
  account_id          = var.dbk-account-id
  azure_client_id     = var.dbk-acct-admin-client-id
  azure_client_secret = var.dbk-acct-admin-client-secret

}

resource "databricks_metastore_assignment" "metastore-assignment" {
  count        = var.skip-metastore-assignment ? 0 : 1
  provider     = databricks.uc-account
  workspace_id = azurerm_databricks_workspace.adb.workspace_id
  metastore_id = lookup(var.metastore-id, var.location, "eastus2")
  depends_on = [
    azurerm_databricks_workspace.adb
  ]
}

```

### Expected Behavior
Workspace should  be assigned to the metastore or metastore assignment should be read.

### Actual Behavior
We get the following error

│ Error: cannot read metastore assignment: invalid character '<' looking for beginning of value
│
│   with module.adb-ws.databricks_metastore_assignment.metastore-assignment[0],
│   on databricks-workspace/module/metastore-assignment.tf line 7, in resource "databricks_metastore_assignment" "metastore-assignment":
│    7: resource "databricks_metastore_assignment" "metastore-assignment" {
│

### Steps to Reproduce

1. Use Azure client Id and Client secrte to AUth.  SPN should be workspace and Metastore Admin.
2. Use an alias provider and spn auth for authenticating to databricks account.
3. use the resource "databricks_metastore_assignment" to assign the metastore.
4. Run terraform Plan/Apply

### Terraform and provider versions
Terraform version: 1.1.6
Databricks provider including 1.24.1

### Debug Output
module.adb-nat-eastus2-ws.databricks_metastore_assignment.metastore-assignment[0]: Refreshing state... [id=1272903653929400|9ae95b0d-75c4-45f4-a44c-b49cf98f46cb]
2023-09-06T12:56:15.732Z [INFO]  provider.terraform-provider-databricks_v1.24.1: Generating AAD token for Service Principal (55958b75-d1ee-4736-8b6b-ed4a3dce6f29): timestamp=2023-09-06T12:56:15.732Z2023-09-06T12:56:16.025Z [ERROR] provider.terraform-provider-databricks_v1.24.1: Response contains error diagnostic: diagnostic_detail= diagnostic_severity=ERROR diagnostic_summary="cannot read metastore assignment: invalid character '<' looking for beginning of value" tf_proto_version=5.3 @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:58 tf_req_id=05ade95d-1c41-9862-f915-49758e957a43 tf_resource_type=databricks_metastore_assignment tf_rpc=ReadResource @module=sdk.proto tf_provider_addr=registry.terraform.io/databricks/databricks timestamp=2023-09-06T12:56:16.025Z
2023-09-06T12:56:16.026Z [ERROR] vertex "module.adb-nat-eastus2-ws.databricks_metastore_assignment.metastore-assignment[0]" error: cannot read metastore assignment: invalid character '<' looking for beginning of value
2023-09-06T12:56:16.026Z [ERROR] vertex "module.adb-nat-eastus2-ws.databricks_metastore_assignment.metastore-assignment" error: cannot read metastore assignment: invalid character '<' looking for beginning of value
2023-09-06T12:56:16.026Z [ERROR] vertex "module.adb-nat-eastus2-ws.databricks_metastore_assignment.metastore-assignment (expand)" error: cannot read metastore assignment: invalid character '<' looking for beginning of value
2023-09-06T12:56:16.026Z [DEBUG] created provider logger: level=debug
2023-09-06T12:56:16.026Z [INFO]  provider: configuring client automatic mTLS
2023-09-06T12:56:16.035Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/vault/3.20.0/linux_amd64/terraform-provider-vault_v3.20.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/vault/3.20.0/linux_amd64/terraform-provider-vault_v3.20.0_x5]
2023-09-06T12:56:16.036Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/vault/3.20.0/linux_amd64/terraform-provider-vault_v3.20.0_x5 pid=268
2023-09-06T12:56:16.036Z [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/vault/3.20.0/linux_amd64/terraform-provider-vault_v3.20.0_x5
2023-09-06T12:56:16.050Z [INFO]  provider.terraform-provider-vault_v3.20.0_x5: configuring server automatic mTLS: timestamp=2023-09-06T12:56:16.050Z
2023-09-06T12:56:16.069Z [DEBUG] provider.terraform-provider-vault_v3.20.0_x5: plugin address: address=/tmp/plugin3160120075 network=unix timestamp=2023-09-06T12:56:16.069Z
2023-09-06T12:56:16.069Z [DEBUG] provider: using plugin: version=5
2023-09-06T12:56:16.093Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2023-09-06T12:56:16.094Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/vault/3.20.0/linux_amd64/terraform-provider-vault_v3.20.0_x5 pid=268      
2023-09-06T12:56:16.094Z [DEBUG] provider: plugin exited
2023-09-06T12:56:16.098Z [INFO]  backend/local: plan operation completed
╷
│ Error: cannot read metastore assignment: invalid character '<' looking for beginning of value
│
│   with module.adb-nat-eastus2-ws.databricks_metastore_assignment.metastore-assignment[0],
│   on databricks-workspace/module/metastore-assignment.tf line 36, in resource "databricks_metastore_assignment" "metastore-assignment":
│   36: resource "databricks_metastore_assignment" "metastore-assignment" {
│

### Important Factoids
I confirmed that the authentication was successful by creating other account level resources such as databicks_groups using the alias provider and it works as expected. this issue is narrowed meatsore assignment.

For eg: This will not have any issues
```
data "databricks_group" "admins" {
  provider     = databricks.uc-account
  display_name = "AZ_adb-APP0000256-efl-npe"
}

```
2023-09-06T13:42:11.028Z [DEBUG] Resource instance state not found for node "data.databricks_group.admins", instance data.databricks_group.admins
2023-09-06T13:42:11.029Z [DEBUG] ReferenceTransformer: "data.databricks_group.admins" references: []
2023-09-06T13:42:11.030Z [INFO]  provider.terraform-provider-databricks_v1.24.1: Generating AAD token for Service Principal (55958b75-d1ee-4736-8b6b-ed4a3dce6f29): timestamp=2023-09-06T13:42:11.030Z2023-09-06T13:42:11.581Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2023-09-06T13:42:11.582Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/databricks/databricks/1.24.1/linux_amd64/terraform-provider-databricks_v1.24.1 pid=221
2023-09-06T13:42:11.582Z [DEBUG] provider: plugin exited
2023-09-06T13:42:11.584Z [INFO]  backend/local: plan operation completed

Few thing:
The Service Principle is an account admin
The SPN is also part of a group that is the metastore owner.

**Comment by tanmay-db** [comment]
Hi @sivadotblog, this should have been resolved in 1.28.1 by the change: https://github.com/databricks/terraform-provider-databricks/pull/2804. Can you confirm if this is still happening on that version or this is resolved?

**Comment by tanmay-db** [comment]
Closing this for now, please feel free to reopen if this is still happening.
